### PR TITLE
Add partial support for iPhone X

### PR DESCRIPTION
**Pull request by erikstrottmann**
Partial support for #48.

## Photo library picker:

Old:

(Automatically scrolls too far, and shows photos between navigation bar and top of screen)

screen shot 2017-10-31 at 01 27 54

New:

screen shot 2017-10-31 at 01 28 31
screen shot 2017-10-31 at 01 28 57

## Camera

Old:

(Buttons and “Photo 1” text too high)

screen shot 2017-10-31 at 01 28 02

New:

screen shot 2017-10-31 at 01 29 17

Same photo on iPhone 6s, for comparison:

screen shot 2017-10-31 at 01 27 33

## Already supporting iPhone X:

Circle cropper:

screen shot 2017-10-31 at 01 43 10

## Still not supporting iPhone X:

Photo library in landscape:

(Photos too close to left and right edges)

screen shot 2017-10-31 at 01 30 19

Camera in landscape:

(Buttons to close to “top” [physical top of phone, not virtual top of screen])

screen shot 2017-10-31 at 01 30 13

Edit tool:

screen shot 2017-10-31 at 01 44 14

### CODE REVIEW

File: Paparazzo/Core/VIPER/MediaPicker/View/MediaPickerView.swift [hunk 1]
```diff
@@ -549,17 +549,24 @@ final class MediaPickerView: UIView, ThemeConfigurable {
         
         let leftButton = closeAndContinueButtonsSwapped ? continueButton : closeButton
         let rightButton = closeAndContinueButtonsSwapped ? closeButton : continueButton
+
+        let topMargin: CGFloat
+        if #available(iOS 11.0, *) {
+            topMargin = safeAreaInsets.top
+        } else {
+            topMargin = 0
+        }
```
Comment (HiveHicks) [inline Paparazzo/Core/VIPER/MediaPicker/View/MediaPickerView.swift]: To avoid checking for the availability of `safeAreaInsets` I would add a `UIView` extension like that:

```swift
extension UIView {
    var paparazzoSafeAreaInsets: UIEdgeInsets {
        if #available(iOS 11.0, *) {
            return safeAreaInsets
        } else {
            return .zero
        }
    }
}
```

File: Paparazzo/Core/VIPER/PhotoLibrary/View/PhotoLibraryViewController.swift [hunk 2]
```diff
@@ -21,8 +21,6 @@ final class PhotoLibraryViewController: PaparazzoViewController, PhotoLibraryVie
         super.viewWillAppear(animated)
         
         navigationController?.setNavigationBarHidden(false, animated: animated)
-        hideNavigationBarShadow()
```
Comment (HiveHicks) [inline Paparazzo/Core/VIPER/PhotoLibrary/View/PhotoLibraryViewController.swift]: What is the purpose of this change?

File: Example/PaparazzoExample.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -43,6 +43,9 @@
 		25A489CC1E656B2D00CC431B /* Latoto-SemiboldItalic.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 251E57EC1E6565890009A288 /* Latoto-SemiboldItalic.ttf */; };
 		2A0B5C0CB29FF72D8233F563 /* Pods_PaparazzoExample_Storyboard.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 0197BFD7D8A9DC3C06FA3EB3 /* Pods_PaparazzoExample_Storyboard.framework */; };
 		38D34B668F9788158C7228EA /* Pods_PaparazzoExample.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 11AE831F8581BCFBA632B260 /* Pods_PaparazzoExample.framework */; };
+		BB0265081FA86FB4002A5561 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = BB0265061FA86FB4002A5561 /* LaunchScreen.storyboard */; };
+		BB0265091FA86FB4002A5561 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = BB0265061FA86FB4002A5561 /* LaunchScreen.storyboard */; };
+		BB02650A1FA86FB4002A5561 /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = BB0265061FA86FB4002A5561 /* LaunchScreen.storyboard */; };
```
[the remark this answers is not in the document]
Comment (erikstrottmann) [reply Example/PaparazzoExample.xcodeproj/project.pbxproj]: Per Apple’s guide to [Update Apps for iPhone X](https://developer.apple.com/ios/update-apps-for-iphone-x/), apps need either a `LaunchScreen.storyboard` or an iPhone X launch image ([1125px × 2436px](https://developer.apple.com/ios/human-interface-guidelines/icons-and-images/launch-screen/)). Since Paparazzo already included the `LaunchScreen.storyboard` for the storyboard-based example project, I moved the storyboard to the shared directory (`Example/PaparazzoExample_Storyboard/Base.lproj/LaunchScreen.storyboard` → `Example/Shared/Base.lproj/LaunchScreen.storyboard`) and included it in the other two targets. This diff hunk shows the file being included in all three targets, not a duplication of the file.

If you prefer not to include even a `LaunchScreen.storyboard` in the non-storyboard-based targets, please provide a 1125px × 2436px launch screen image.

File: Paparazzo/Core/Extensions/UIScrollView+ScrollToBottom.swift [hunk 4]
```diff
@@ -5,10 +5,17 @@ extension UIScrollView {
     func scrollToBottom() {
         
         layoutIfNeeded()
-        
+
+        let minimumYOffset: CGFloat
+        if #available(iOS 11.0, *) {
+            minimumYOffset = -safeAreaInsets.top
```
[the remark this answers is not in the document]
Comment (erikstrottmann) [reply Paparazzo/Core/Extensions/UIScrollView+ScrollToBottom.swift]: Agreed.

File: Paparazzo/Core/VIPER/MediaPicker/View/MediaPickerView.swift (hunk 1 above)
Comment (erikstrottmann) [reply Paparazzo/Core/VIPER/MediaPicker/View/MediaPickerView.swift]: Agreed.

File: Paparazzo/Core/VIPER/PhotoLibrary/View/PhotoLibraryViewController.swift (hunk 2 above)
Comment (erikstrottmann) [reply Paparazzo/Core/VIPER/PhotoLibrary/View/PhotoLibraryViewController.swift]: One of the navigation bar changes in `hideNavigationBarShadow` causes the table view’s contents to appear in the iPhone X status bar. (See the first image from this PR, reproduced below.) I’m not sure whether it’s setting the background image, the background color, or the shadow image. If you prefer to still set some of those, while removing the offending ones, some experimentation could determine which are causing the visual error below.

screen shot 2017-10-31 at 01 27 54

**Comment by erikstrottmann** [comment]
I can make the requested changes, but won’t be able to get back to this for a few days. If you’d like the changes sooner, this PR is actually already allows edits from maintainers of this project:

allowed-edits-from-maintainers

File: Example/PaparazzoExample.xcodeproj/project.pbxproj (hunk 3 above)
[the remark this answers is not in the document]
Comment (HiveHicks) [reply Example/PaparazzoExample.xcodeproj/project.pbxproj]: Oh, I didn't know that, thanks for the explanation!
